src/data/data_file_handler.py

```diff
@@ -13,27 +13,26 @@
 
 
 class DataFileHandler(DataHandler):
-    def __init__(self):
-        super().__init__()
-        self._dataframe = pd.DataFrame()
+    def __init__(self, ignore: Optional[List] = None):
+        super().__init__(ignore)
 
-    def add_dataframes_from_folders_as_mean(self, folders: List, points: int):
+    def concat_dataframes_from_folders_as_mean(self, folders: List, points: int):
         for folder in folders:
             folder_path = Path(folder)
             folder_df = pd.DataFrame()
             match file_utils.get_files_from_folder_path(folder_path):
                 case Ok(data_files_names):
-                    folder_df = self._add_files_as_mean(folder_df, data_files_names, folder_path,
-                                                        points)
+                    folder_df = self._concat_files_as_mean(folder_df, data_files_names, folder_path,
+                                                           points)
                     folder_df["class"] = f"{folder_path.parent.name}/{folder_path.name}"
                     self._dataframe = pd.concat([self._dataframe, folder_df])
                 case Err(error):
                     logger.error(error)
 
-    def _add_files_as_mean(self, target_df: pd.DataFrame, data_files_path: List,
-                           location: Path, points: int) -> pd.DataFrame:
+    def _concat_files_as_mean(self, target_df: pd.DataFrame, data_files_path: List,
+                              location: Path, points: int) -> pd.DataFrame:
         for file_path in data_files_path:
-            match self.add_data_from_file(target_df, location / file_path, points):
+            match self._concat_data_as_mean_from_file(target_df, location / file_path, points):
                 case Ok(result_df):
                     logger.info(f"{file_path} added")
                     target_df = result_df
```

Comment:
On the other hand, can the `target_df = result_df` be changed to a `return result_df`? Or do we want to only return the last value assigned?

Reply:
I also don't follow you here.
If we change `target_df = result_df` to a `return result_df` the for loop will not be completed, and we will return only the first `file_path` appended.
This is not returning the last value assigned, it is appending all values from all `data_files_path` into `target_df`.

Reply:
Answered in the other comment.